### ISSUE TITLE
url path parse bug

### ISSUE THREAD

**Issue by Billskate** [janhommes/o.js#10]
I use TripPinService of OData.org to test, but I found the request's url is wrong.

Here is the test: https://jsfiddle.net/billskate/su3b32mv/1/

The correct url is **http://services.odata.org/V4/(S(ms4wufavzmwsg3fjo3eqdgak))/TripPinServiceRW/People**,
but parsed to
**http://services.odata.org/V4//TripPinServiceRW/People/**

This part **(S(ms4wufavzmwsg3fjo3eqdgak))** is missed.

Another bug is you can not add **Content-Length** header to XMLHttpRequest.

**Comment by janhommes** [comment]
Yep, that should not happen. As long I am working on a fix you can use the endpoint configuration:

```
o().config({
			endpoint:'http://services.odata.org/V4/%28S%28wptr35qf3bz4kb5oatn432ul%29%29/TripPinServiceRW/',
			version:4,
			strictMode:true
		});

````

And than patch your data without the full url:
```
o('People').post({
  UserName: 'billskate',
  FirstName: 'Bill',
  LastName: 'Skate'
}).save();
```

**Comment by Billskate** [comment]
thx

**Comment by janhommes** [comment]
fixed in 2e875ee
